docs/adr/0002-yarn-v4.md

```diff
@@ -0,0 +1,243 @@
+# Extend Yarn Berry support to v4
+
+## Context
+
+Yarn v4 was released in October 23, 2023, which coincided with the time that the support for Yarn v3 was being introduced in Cachi2. To limit the scope of implementation, we decided to leave support for v4 to a later time. Now, as more projects migrate to v4, introducing proper support has become a priority.
+
+This document analyzes the the most impactful changes introduced with v4, and how they relate to the current v3 implementation.
+
+Main references:
+- [v4 blog post](https://yarnpkg.com/blog/release/4.0)
+- [Breaking changes for Yarn 4](https://github.com/yarnpkg/berry/issues/3591)
+
+### Global cache is enabled by default
```

Comment:
I like it, but then we're adding a new header that is not present in the standard [ADR template](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions).

I wonder if we should move this section to another document, something like a `docs/designs` folder, that can be supplemental to the ADRs.

Or... we can simply take liberties with the original proposed ADR template, not sure what's better.

Reply:
This document is a fine place to keep the section, having it here does not introduce discontinuity. I'd say having an extra section that improves the flow of the doc is better than obedience to the template (but this is a debatable point).

What about going one level deeper then?

```diff
+ ### Changes in Yarn behavior
+ #### Global cache is enabled by default
```

Reply:
I made it one level of nesting deeper, but I don't really like the result. It is fine since this is a small investigation, but would probably be troublesome for larger docs. We can probably add extra sections in those cases, I see no reason why we need to be extra strict with the original template.

Reply:
Thank you. I think extra sections are fine, and the nesting is mostly fine too.